compiler/src/dotty/tools/dotc/core/SymbolLoaders.scala

```diff
@@ -432,10 +432,9 @@ class TastyLoader(val tastyFile: AbstractFile) extends SymbolLoader {
 
 
   private def checkTastyUUID(tastyFile: AbstractFile, tastyBytes: Array[Byte])(using Context): Unit =
-    var classfile = tastyFile.resolveSibling(tastyFile.name.stripSuffix(".tasty") + ".class")
-    if classfile == null then
-      classfile = tastyFile.resolveSibling(tastyFile.name.stripSuffix(".tasty") + "$.class")
-    if classfile != null then
+    import dotty.tools.io.NoAbstractFile
+    val classfile = ctx.getSiblingClassfile(tastyFile)
```

Comment:
How much does this cache help? I would expect we read this file a single time. If that is the case, could we make the cache local to `ExtractDependencies`?

Reply:
we have to read it at least once when we load the class symbol, so why waste time resolving the second time, especially as it does File IO